src/main/java/org/opensearch/knn/index/util/Lucene.java

```diff
@@ -44,6 +71,10 @@ public class Lucene extends JVMLibrary {
                         v -> v > 0
                     )
                 )
+                .addParameter(
+                    METHOD_ENCODER_PARAMETER,
+                    new Parameter.MethodComponentContextParameter(METHOD_ENCODER_PARAMETER, null, HNSW_ENCODERS)
```

Comment:
Why null here as well?

Reply:
Lucene doesn't have any default encoder like faiss. Should I set it to HNSW_ENCODERS ?

Reply:
For faiss, I set it to the flat encoding awhile ago. I just worry a little bit around NPE. Can we have a flat encoder be a noop?

Reply:
Sure